iris_grib/_load_convert.py

```diff
@@ -843,24 +843,34 @@ def grid_definition_template_12(section, metadata):
     # Check whether Di and Dj are as consistent as possible with that
     # interpretation - i.e. they are within 1cm.
     def check_range(v1, v2, n, d):
-        min_last = v1 + (n - 1) * (d - 1)
-        max_last = v1 + (n - 1) * (d + 1)
-        if not (min_last < v2 < max_last):
+        small = min(v1, v2)
+        large = max(v1, v2)
+        min_last = small + (n - 1) * (d - 1)
+        max_last = small + (n - 1) * (d + 1)
+        if not (min_last < large < max_last):
             raise TranslationError('Inconsistent grid definition')
     check_range(x1, x2, section['Ni'], section['Di'])
     check_range(y1, y2, section['Nj'], section['Dj'])
 
+    # Further over-specification - the sequence of X1 & X2 is enough to
+    #  generate the sequence in the correct direction (also Y1 & Y2). All
+    #  scanningMode can do is add confusion; warn if there is inconsistency.
+    def validate_scanning(axis: str, stated: bool, encoded: bool):
+        def scan_str(scanning_bool):
+            return "positive" if scanning_bool else "negative"
+        if stated != encoded:
+            message = (
+                f"File states {scan_str(stated)} {axis} scanning, but encodes "
```

Comment:
This message could also be clearer :  IMHO it isn't clear what 'states' and 'encoded' mean.
Suggest messages more like 
"Gridpoint direction doesn't match X scanning mode : 'X2' > 'X1', but 'i_negative'=True."

I think/hope that can be generalised fairly easily?

Reply:
c2e482698ed5f115121db8530c8a7e1bb1d90848